twill-common/src/main/java/org/apache/twill/filesystem/LocalLocation.java

```diff
@@ -251,6 +251,34 @@ public boolean mkdirs() throws IOException {
     return file.mkdirs();
   }
 
+  @Override
+  public boolean mkdirs(String permission) throws IOException {
```

Comment:
hmm that does not seem to apply the permissions correctly, it applies the umask to the permissions. What we want is the exact permissions as passed in.